### PR TITLE
Calculator: [Phase 3] Add History / Calc reloading

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Implements Phase 3 - Ledger (History). Reloading + Calculation continuation available.

screen shot 2017-04-11 at 10 55 43 pm

## Related Issues and Discussions

Continues: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/4067
Fixes: #4066

## People to notify

@moollaza 
@bsstoner 
@russellholt 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Calculator.pm [hunk 1]
```diff
@@ -201,7 +201,8 @@ sub prepare_for_display {
             templates => {
                 group => 'base',
                 options => {
-                    content => 'DDH.calculator.content'
+                    content => 'DDH.calculator.content',
+                    ledger_item => 'DDH.calculator.ledger_item'
```
Comment (moollaza) [inline lib/DDG/Goodie/Calculator.pm]: This line isn't necessary. It's only for defining options for the `base` template, which expects a `content` variable to be defined.

Setting `ledger_item` has no effect. We automatically pre-compile all the templates in the share dir and make them available 👍

File: share/goodie/calculator/calculator.js [hunk 2]
```diff
@@ -397,21 +397,31 @@ DDH.calculator = DDH.calculator || {};
...
+        addToHistory: function( expression, result ) {
+            // DDH.calculator.ledger_item is a ref to the ledger_item.handlebars template
+            var ledger_item = DDH.calculator.ledger_item({expression: expression, result: result});
+            $(".tile__history").prepend(ledger_item);
+        },
+        
+        // reloadIntoCalc: reloads the clicked <li> into the calculator
+        reloadIntoCalc: function( expression, result ) {
+            ExpressionParser.setExpression(expression);
+            display.value = result.replace(/,/, '');
```
Comment (moollaza) [inline share/goodie/calculator/calculator.js]: need `/g` flag on the regex I believe?

File: lib/DDG/Goodie/Calculator.pm (hunk 1 above)
Comment (pjhampton) [inline lib/DDG/Goodie/Calculator.pm]: I didn't know that :) thanks!

File: share/goodie/calculator/calculator.js (hunk 2 above)
Comment (pjhampton) [inline share/goodie/calculator/calculator.js]: Super nice catch!

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Pull request merged**